src/lib/operators/index_scan.cpp

```diff
@@ -36,30 +38,69 @@ std::shared_ptr<const Table> IndexScan::_on_execute() {
 
   _validate_input();
 
+  auto chunk_id_mapping = std::vector<std::optional<ChunkID>>{};
+  if (!_chunk_id_mapping) {
+    chunk_id_mapping = std::vector<std::optional<ChunkID>>(_in_table->chunk_count());
+    std::iota(chunk_id_mapping.begin(), chunk_id_mapping.end(), ChunkID{0});
+  } else {
+    chunk_id_mapping = *_chunk_id_mapping;
+  }
+
   _out_table = std::make_shared<Table>(_in_table->column_definitions(), TableType::References);
+  auto matches_out = std::make_shared<RowIDPosList>();
 
-  std::mutex output_mutex;
+  const auto& indexes = _in_table->get_table_indexes(_indexed_column_id);
+  Assert(!indexes.empty(), "No indexes for the requested ColumnID available.");
 
-  auto jobs = std::vector<std::shared_ptr<AbstractTask>>{};
-  if (included_chunk_ids.empty()) {
-    const auto chunk_count = _in_table->chunk_count();
-    jobs.reserve(chunk_count);
-    for (auto chunk_id = ChunkID{0u}; chunk_id < chunk_count; ++chunk_id) {
-      const auto chunk = _in_table->get_chunk(chunk_id);
-      Assert(chunk, "Physically deleted chunk should not reach this point, see get_chunk / #1686.");
+  Assert(indexes.size() == 1, "We do not support the handling of multiple indexes for the same column.");
+  const auto& index = indexes.front();
 
-      jobs.push_back(_create_job(chunk_id, output_mutex));
-    }
-  } else {
-    jobs.reserve(included_chunk_ids.size());
-    for (const auto chunk_id : included_chunk_ids) {
-      if (_in_table->get_chunk(chunk_id)) {
-        jobs.push_back(_create_job(chunk_id, output_mutex));
+  const auto append_matches = [&](const auto& begin, const auto& end) {
+    for (auto current_iter = begin; current_iter != end; ++current_iter) {
+      // If the matches Chunk is included, its mapped ChunkID is added to the output. The ChunkOffset stays the same.
+      if (binary_search(included_chunk_ids.begin(), included_chunk_ids.end(),
```

Comment:
with the suggestion above, this would be:
```c++
if (!pruned_chunk_ids.contains(current_iter->chunk_id)) {
  ...
}
```
Right? You don't need the mapping, or do I miss something?

Reply:
Pruned ChunkIDs are not added to the ìncluded_chunk_ids` by the `LQPTranslator`, so it is not necessary to filter them out here.
But as I said **we receive wrong results** when we do not map the ChunkIDs received from the TableIndex to the pruned ChunkIDs.